### PR TITLE
KOGITO-4614: add dmn-knative-quickstart-quarkus

### PR DESCRIPTION
**Pull request by kostola**
- [Jira Task](https://issues.redhat.com/browse/KOGITO-4614)
- [Related PR in kogito-runtimes](https://github.com/kiegroup/kogito-runtimes/pull/1412)

### CODE REVIEW

File: dmn-knative-quickstart-quarkus/README.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+```sh
+K_SINK=http://localhost:8181 mvn clean compile quarkus:dev
+```
+
+[`K_SINK` is the environment variable injected by the Knative Eventing platform](https://knative.dev/docs/eventing/samples/sinkbinding/#create-our-sinkbinding)
+once we deploy the application on a Kubernetes cluster. Instead of _hardcoding_ the endpoint where we are going to send our produced messages, we rely
+on Knative to inject this information in runtime.
+
+The environment variable will be assigned to the SmallRye HTTP connector in runtime: `mp.messaging.outgoing.processedtravellers.url=${K_SINK}`.
```
Comment (ricardozanini) [inline dmn-knative-quickstart-quarkus/README.md]:
```suggestion
The environment variable will be assigned to the Quarkus HTTP connector in runtime: `mp.messaging.outgoing.processedtravellers.url=${K_SINK}`.
```

Comment (ricardozanini) [reply dmn-knative-quickstart-quarkus/README.md]: `processedtravellers` is this correct?

**Review by ricardozanini — APPROVED** [review]
Really good! Just take a look at the docs to make sure the copy and paste from the other examples are ok. :sweat_smile: 

I wanna see this example with this: https://github.com/ricardozanini/eventing-kogito/pull/1/

File: dmn-knative-quickstart-quarkus/README.md (hunk 1 above)
Comment (kostola) [reply dmn-knative-quickstart-quarkus/README.md]: Good catch! Fixed :+1:

**Comment by ricardozanini** [comment]
jenkins rerun examples tests

File: dmn-knative-quickstart-quarkus/src/test/java/org/kie/kogito/examples/CloudEventListenerTest.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                .header("ce-kogitodmnmodelname", "Traffic Violation")
+                .header("ce-kogitodmnmodelnamespace", "https://github.com/kiegroup/drools/kie-dmn/_A4BCA8B8-CF08-433F-93B2-A2598F19ECFF")
+                .contentType(MediaType.APPLICATION_JSON)
+                .body(new ObjectMapper().writeValueAsString(decisionInput)).post("/").then().statusCode(200);
+
+        await()
+                .atLeast(2, SECONDS)
+                .atMost(10, SECONDS)
+                .with().pollInterval(2, SECONDS)
+                .untilAsserted(() -> sink.verify(1, postRequestedFor(urlEqualTo("/"))));
```
Comment (danielezonca) [inline dmn-knative-quickstart-quarkus/src/test/java/org/kie/kogito/examples/CloudEventListenerTest.java]: Can you add some assertion on the actual payload of the result?

File: dmn-knative-quickstart-quarkus/pom.xml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      <scope>test</scope>
+    </dependency>
+    <!-- required to mock the sink server -->
+    <!-- Unfortunately MockWebServer is incompatible with RestEasy:
+    java.lang.NoSuchMethodError: 'com.github.fge.jackson.JsonNumEquals com.github.fge.jackson.JsonNumEquals.getInstance()'
+    -->
+    <dependency>
+      <groupId>com.github.tomakehurst</groupId>
+      <artifactId>wiremock-jre8</artifactId>
+      <version>${version.com.github.tomakehurst.wiremock}</version>
```
Comment (danielezonca) [inline dmn-knative-quickstart-quarkus/pom.xml]: You should have this dependency already in [dependency management](https://github.com/kiegroup/kogito-runtimes/blob/master/kogito-build/kogito-build-parent/pom.xml#L460)

File: dmn-knative-quickstart-quarkus/src/test/java/org/kie/kogito/examples/CloudEventListenerTest.java (hunk 2 above)
Comment (kostola) [reply dmn-knative-quickstart-quarkus/src/test/java/org/kie/kogito/examples/CloudEventListenerTest.java]: Done :+1:

File: dmn-knative-quickstart-quarkus/pom.xml (hunk 3 above)
Comment (kostola) [reply dmn-knative-quickstart-quarkus/pom.xml]: Done :+1:

**Review by danielezonca — APPROVED** [review]

**Pull request merged**